dev/benchmarks/macrobenchmarks/lib/src/web/bench_text_layout.dart

```diff
@@ -2,37 +2,147 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:math';
 import 'dart:ui';
 
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart';
+
 import 'recorder.dart';
 
-int _counter = 0;
+const MethodCodec codec = JSONMethodCodec();
 
-Paragraph _generateParagraph() {
-  final ParagraphBuilder builder =
-      ParagraphBuilder(ParagraphStyle(fontFamily: 'sans-serif'))
-        ..pushStyle(TextStyle(fontSize: 12.0))
-        ..addText(
-          '$_counter Lorem ipsum dolor sit amet, consectetur adipiscing elit, '
-          'sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
-        );
-  _counter++;
-  return builder.build();
+const String chars = '1234567890'
+    'abcdefghijklmnopqrstuvwxyz'
+    'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
+    '!@#%^&()[]{}<>,./?;:"`~-_=+|';
+
+String _randomize(String text) {
+  return text.replaceAllMapped(
+    '*',
+    (_) => chars[Random().nextInt(chars.length)],
```

Comment:
If you are going to use `Random` consider giving it a constant seed, e.g. `Random(0)`. Otherwise, the results won't be reproducible. For the other text benchmark I got sufficient randomness by picking words out of lipsum without using `Random`.